GetIntoTeachingApi/Models/TeachingEventRegistration.cs

```diff
@@ -33,5 +33,17 @@ public TeachingEventRegistration(Entity entity, ICrmService crm)
             : base(entity, crm)
         {
         }
+
+        protected override bool ShouldMap(ICrmService crm)
+        {
+            var alreadyRegistered = !crm.CandidateYetToRegisterForTeachingEvent(CandidateId, EventId);

```

Comment:
dont need the variable here do we?

if(!thething())
{
    return false;
}

Reply:
It was just to make it more readable as it's negating the function call - I could have reworked the function to something like `crm.CandidateAlreadyRegisteredForTeachingEvent` but there are multiple in this format so opted to keep them consistent (and the current method name makes more sense in the other use cases).